ir/pattern/optional.rs

```diff
@@ -40,6 +41,24 @@ impl Optional {
     pub fn conjunction_mut(&mut self) -> &mut Conjunction {
         &mut self.conjunction
     }
+
+    pub(crate) fn variable_dependency(
+        &self,
+        block_context: &BlockContext,
+    ) -> HashMap<Variable, VariableDependency<'_>> {
+        self.conjunction
+            .variable_dependency(block_context)
+            .into_iter()
+            .map(|(var, mut mode)| {
+                // VariableDependency::Producing means "producing in all branches".
```

Comment:
Is this even relevant anymore, we don't treat optionals as branches i thought

Reply:
"In all code paths" if you prefer.

Reply:
But why do you have a comment on what VariableDependency means here of all places? I didn't see something similar in disjunctions or negations

Reply:
Because here we're unconditionally overriding all "producing" modes and I figured that the first question would be "why?".

Reply:
Right, maybe answer that directly then? `A try block never produces output, and can only have local or input variables`?

Is that we've agreed on for now? I think we will revisit this anyway

Reply:
Good idea to leave a comment tho